app/controllers/journeys/TaskListController.scala

```diff
@@ -52,24 +55,51 @@ class TaskListController @Inject() (override val messagesApi: MessagesApi,
   def onPageLoad(taxYear: TaxYear): Action[AnyContent] = (identify andThen getData) async { implicit originalRequest =>
     val result = for {
       taskListWithRequest <- answerLoader.loadTaskList(taxYear, originalRequest)
-      taskList              = taskListWithRequest.taskList
-      updatedRequest        = taskListWithRequest.request
-      completedTrades       = getViewModelList(taskList)
-      message               = messagesApi.preferred(updatedRequest)
-      tradeDetailsStatus    = taskList.tradeDetails.map(_.journeyStatus).getOrElse(CheckOurRecords)
-      idsWithAccountingType = completedTrades.map(t => (t.tradingName.getOrElse(TradingName.empty), t.accountingType, t.businessId))
-      updatedUserAnswers <- EitherT.right[ServiceError](service.setAccountingTypeForIds(updatedRequest.answers, idsWithAccountingType))
-      viewModelList             = completedTrades.map(TradesJourneyStatuses.toViewModel(_, taxYear, updatedUserAnswers.some)(message))
-      nationalInsuranceStatuses = taskList.nationalInsuranceContributions
-      nationalInsuranceSummary = NationalInsuranceContributionsViewModel.buildSummaryList(nationalInsuranceStatuses, completedTrades, taxYear)(
-        message)
-    } yield Ok(view(taxYear, updatedRequest.user, tradeDetailsStatus, viewModelList, nationalInsuranceSummary)(updatedRequest, message))
+      updatedRequest     = taskListWithRequest.request
+      messages           = messagesApi.preferred(updatedRequest)
+      tradeDetailsStatus = getTradeDetailsStatus(taskListWithRequest)
+      completedTrades    = getTradesIfDetailsAreCompleted(taskListWithRequest.taskList)
+      businessSummaryList      <- getBusinessSummaries(completedTrades, updatedRequest, taxYear, messages)
+      nationalInsuranceSummary <- getNationalInsuranceSummary(taskListWithRequest, completedTrades, taxYear, messages)
+    } yield Ok(view(taxYear, updatedRequest.user, tradeDetailsStatus, businessSummaryList, nationalInsuranceSummary)(updatedRequest, messages))
     handleResultT(result)
```

Comment:
Tried to break up this huge for comprehension and make it a bit more descriptive